app/navigation/Navigator.scala

```diff
@@ -219,7 +219,12 @@ class Navigator @Inject() (categorisationService: CategorisationService) {
       .get(RemoveNirmsPage)
       .map {
         case true  => routes.CyaMaintainProfileController.onPageLoadNirms
-        case false => routes.ProfileController.onPageLoad()
+        case false =>
+          if (answers.get(NirmsNumberUpdatePage).isDefined) {
+            routes.CyaMaintainProfileController.onPageLoadNirmsNumber
```

Comment:
Scenario where it ends up in here could be tested

Reply:
I shall take a look in a moment :)